js/src/views/Status/NodeStatus/store.js

```diff
@@ -37,37 +41,65 @@ export default class StatusStore {
     this.api = api;
   }
 
-  @action setLongStatus ({ defaultExtraData, enode, netPort, rpcSettings }) {
-    transaction(() => {
-      this.defaultExtraData = defaultExtraData;
-      this.enode = enode;
-      this.netPort = netPort;
-      this.rpcSettings = rpcSettings;
-    });
-  }
-
-  @action setStatus ({ hashrate }) {
-    transaction(() => {
-      this.hashrate = hashrate;
-    });
+  @action setStatuses ({ chain, defaultExtraData, enode, netPeers, netPort, rpcSettings, hashrate }) {
+    this.chain = chain;
+    this.defaultExtraData = defaultExtraData;
+    this.enode = enode;
+    this.netPeers = netPeers;
+    this.netPort = netPort;
+    this.rpcSettings = rpcSettings;
+    this.hashrate = hashrate;
   }
 
   @action setMinerSettings ({ coinbase, extraData, gasFloorTarget, minGasPrice }) {
-    transaction(() => {
-      this.coinbase = coinbase;
-      this.extraData = extraData;
-      this.gasFloorTarget = gasFloorTarget;
-      this.minGasPrice = minGasPrice;
-    });
+    this.coinbase = coinbase;
+    this.extraData = extraData;
+    this.gasFloorTarget = gasFloorTarget;
+    this.minGasPrice = minGasPrice;
   }
 
   startPolling () {
-    this._pollStatus();
-    this._pollLongStatus();
+    this._startPolling();
   }
 
   stopPolling () {
-    Object.keys(this._timeoutIds).forEach((key) => clearTimeout(this._timeoutIds[key]));
+    if (this.subscriptionId) {
+      this.api.pubsub.unsubscribe(this.subscriptionId);
+    }
+  }
+
+  _startPolling () {
+    this.api.pubsub.parity.getBlockHeaderByNumber((err, block) => {
+      if (err) {
+        return;
+      }
+      this.blockNumber = block.number;
+      this.blockTimestamp = block.timestamp;
+      this._pollMinerSettings();
+      return Promise
```

Comment:
`return` here is misleading, it looks a bit like `then(subscriptionId)` is chained to this promise, but it's not.